src/core/MaverickV2.sol

```diff
@@ -206,27 +286,18 @@ abstract contract MaverickV2 is SettlerAbstract {
         uint256 minBuyAmount
     ) internal returns (uint256 buyAmount) {
         bytes memory swapCallbackData = _encodeSwapCallback(permit, sig);
-        address pool = AddressDerivation.deriveDeterministicContract(maverickV2Factory, salt, maverickV2InitHash);
-        (, buyAmount) = abi.decode(
-            _setOperatorAndCall(
-                pool,
-                IMaverickV2Pool(pool).fastEncodeSwap(
-                    recipient,
-                    _permitToSellAmount(permit),
-                    tokenAIn,
-                    false,
-                    tokenAIn.ternary(type(int32).max, type(int32).min),
-                    swapCallbackData
-                ),
-                uint32(IMaverickV2SwapCallback.maverickV2SwapCallback.selector),
-                _maverickV2Callback
-            ),
-            (uint256, uint256)
+        IMaverickV2Pool pool =
+            IMaverickV2Pool(AddressDerivation.deriveDeterministicContract(maverickV2Factory, salt, maverickV2InitHash));
+
+        return _sellToMaverickV2(
+            pool,
+            recipient,
+            tokenAIn,
+            _permitToSellAmount(permit),
+            minBuyAmount,
+            swapCallbackData,
+            _callMaverickWithCallback
```

Comment:
The use of a function pointer here is clever, but unfortunately due to preparations made in solc for the eventual EVM EOF, the gas overhead is considerable (500-600 gas). The only reason we tolerate this patterin in `_setOperatorAndCall` is that there isn't another good way to achieve that functionality without completely blowing out the contract size.